src/scripts/underground/Mine.ts

```diff
@@ -35,30 +33,60 @@ class Mine {
         Mine.grid = tmpGrid;
         Mine.rewardGrid = tmpRewardGrid;
 
+        let Diff = 0;
+        let Added = 0;
         for (let i = 0; i < App.game.underground.getMaxItems(); i++) {
             const item = UndergroundItem.getRandomItem();
-            const x = Mine.getRandomCoord(this.sizeX, item.space[0].length);
-            const y = Mine.getRandomCoord(this.sizeY, item.space.length);
+            const x = Mine.getRandomCoord(Underground.sizeX, item.space[0].length);
+            const y = Mine.getRandomCoord(App.game.underground.getNewYLayer(), item.space.length);
             const res = Mine.canAddReward(x, y, item);
             if (res) {
                 Mine.addReward(x, y, item);
+                Added = Added + 1;
             }
         }
+
+        if (App.game.underground.getMaxItems() < App.game.underground.getMinItems()) {
+            Diff = App.game.underground.getMinItems() - App.game.underground.getMaxItems();
+            //Late-Game players might do some cheesy shit and upgrade min to be above their max. This makes sure it deducts the difference.
+        }
+        for (Added; Added < App.game.underground.getMinItems() - Diff;) {
+            const item = UndergroundItem.getRandomItem();
+            const x = Mine.getRandomCoord(Underground.sizeX, item.space[0].length);
+            const y = Mine.getRandomCoord(App.game.underground.getNewYLayer(), item.space.length);
+            const res = Mine.canAddReward(x, y, item);
+            if (res) {
+                Mine.addReward(x, y, item);
+                Added = Added + 1;
+                //This should loop until it's added.
+            }
+        }
```

Comment:
Have you tested that it's always possible to add the max amount of min items?
As otherwise this could cause an endless loop.

Reply:
Yeah I tested this quite a bit, had the same concerns. 

I found that nine times out of ten, it's not even placing the minimum amount, it's almost always higher than the min. So I don't think will be an issue. If we really wanted to, we could force it to a limit of 100 max attempts to prevent this possibility.

Reply:
I had this backwards, nine times out of ten it's placing the minimum. And it's very, very rarely actually placing the max. 7 without this upgrade and 8 with. I'll post more detailed statistics in a comment of changes. 

I actually think it's impossible for this occur in all situations so marking this as resolved with no changes made. With the Heart Scale as well, it'll find it's way eventually. That being said, it's instant from all the tests i've ran so it's running for long, there's no delay or lag in placing items. If we increase the min limit, I would suggest looking at max 100 attempts as suggested but don't think it's necessary right now.